proxy-robots-module/src/main/java/orwell/proxy/GameState.java

```diff
@@ -0,0 +1,77 @@
+package orwell.proxy;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import orwell.messages.ServerGame;
+import orwell.proxy.robot.RobotGameStateVisitor;
+
+/**
+ * Created by Michaël Ludmann on 6/4/15.
+ */
+public class GameState {
+    private final static Logger logback = LoggerFactory.getLogger(GameState.class);
+    private ServerGame.GameState serverGameGameState;
+
+    public GameState(final byte[] gameStateMessage) {
+        try {
+            this.serverGameGameState = ServerGame.GameState.parseFrom(gameStateMessage);
```

Comment:
Yes, it is allowed to be null (in case of a ParseError, which might happen). In this case, the visitor will be ignored by the class visited. Cf RobotsMap :+1:     @Override
    public void accept(final RobotGameStateVisitor visitor) {
        if (null != visitor)
            visitor.visit(this);
    }